### ISSUE TITLE
Add support to force autogenerated classes to be internal

### ISSUE THREAD

**Issue by tiesmaster** [reactiveui/refit#480]
We started to use Refit, which is really nice, however, it also adds auto generated clients in the NuGet package. Is there a way to exclude it from the assembly, and thus the NuGet package? Since it clobbers the assembly's namespace.

We're using Refit in both a webservice client package, which we distribute as a NuGet package internally. The target framework is the full .NET Framework. And so are the consumer projects that have the package installed, they also have Refit installed, obviously. In the client, I instantiate the clients using `RestService.For<T>(hostUrl)`, so I would expect that the generated clients would only be needed in the consumer projects, and not in the NuGet package, or does Refit not work like that?

**Comment by xljiulang** [comment]
you can use [WebApiClient.JIT](https://github.com/dotnetcore/WebApiClient)，a library that uses Emit to create proxy classes, instead of inserting code at compile time

**Comment by onovotny** [comment]
@tiesmaster Not sure I'm following the issue. At compile-time, Refit looks at the directly referenced project and generates an internal implementation class for the interface. It shouldn't be clobbering any namespaces though. This is not runtime assembly generation.

What issues are you seeing?

**Comment by tiesmaster** [comment]
I've made my Refit-interfaces public, but want my autogenerated classes to be internal. PR #168 made them honor the modifier of the interface, but I'd like them always to be internal. Maybe, with an attribute, like the other Refit-stuff. Per type, or per assembly (my preference).

BTW Initially, I didn't knew _how_ Refit worked, so therefore the initial title of this issue. I've changed that to reflect the real request ;)
